Tasks/Gradle/gradletask.ts

```diff
@@ -151,16 +151,19 @@ async function run() {
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+        // Set working directory
+        let workingDirectory: string = tl.getPathInput('cwd', false, true);
+        if (workingDirectory) {
+            tl.cd(workingDirectory);
+        }
+		
         // Configure wrapperScript
         let wrapperScript: string = tl.getPathInput('wrapperScript', true, true);
         wrapperScript = configureWrapperScript(wrapperScript);
 
-        // Set working directory
-        let workingDirectory: string = tl.getPathInput('cwd', false, true);
         if (!workingDirectory) {
             workingDirectory = path.dirname(wrapperScript);
         }
```

Comment:
Here in the `if (!workingDirectory)` block, we still need `tl.cd(workingDirectory)` after `workingDirectory = path.dirname(wrapperScript);`